packages/node-provider/src/types.ts

```diff
@@ -1,12 +1,25 @@
+export type NodeMessageReceivedCallback = (message: NodeMessage) => void;
+
 export interface INodeProvider {
-  onMessage(callback: (message: NodeMessage) => void);
+  status: NodeProviderStatus;
+  onMessage(callback: NodeMessageReceivedCallback);
   postMessage(message: NodeMessage);
+  connect(): Promise<INodeProvider>;
+  on(messageType: NodeMessageType, callback: NodeMessageReceivedCallback);
+  once(messageType: NodeMessageType, callback: NodeMessageReceivedCallback);
+}
+
+export enum NodeProviderStatus {
+  OFFLINE = "offline",
+  CONNECTED = "connected"
 }
 
 export enum NodeMessageType {
   INSTALL = "install",
   QUERY = "query",
   ERROR = "error"
+  // Should we add a STATUS_CHANGED type to properly
```

Comment:
At the PoC work, we added `cf-node-provider:init` and `cf-node-provider:ready` internal messages to configure the MessagePort instances required to connect the Playground to the iframe. I think `NodeMessageType` refers only to Node-specific messages. How would we handle these _handshake_ messages?

Reply:
Those are strictly internal messages for the playground and the NodeProvider to coordinate over postMessage. I don't think they need to be typed.

Reply:
I think they should be typed but not exposed. It's almost always valuable to have type checks.

Reply:
they are implementation details of how we setup the communication between the iframe and parent frame. it was a misunderstanding to even bring up the idea of typing them as they are just strings passed to postMessage which is a DOM built in which takes any